src/callbacks_stage/subcell_limiter_idp_correction.jl

```diff
@@ -0,0 +1,68 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+"""
+    SubcellLimiterIDPCorrection()
+
+Perform antidiffusive correction stage for the a posteriori IDP limiter [`SubcellLimiterIDP`](@ref)
+called with [`VolumeIntegralSubcellLimiting`](@ref).
+
+!!! note
+    This callback and the actual limiter [`SubcellLimiterIDP`](@ref) only work together.
+    This is not a replacement but a necessary addition.
+
+## References
+
+- Rueda-Ramírez, Pazner, Gassner (2022)
+  Subcell Limiting Strategies for Discontinuous Galerkin Spectral Element Methods
+  [DOI: 10.1016/j.compfluid.2022.105627](https://doi.org/10.1016/j.compfluid.2022.105627)
+- Pazner (2020)
+  Sparse invariant domain preserving discontinuous Galerkin methods with subcell convex limiting
+  [DOI: 10.1016/j.cma.2021.113876](https://doi.org/10.1016/j.cma.2021.113876)
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct SubcellLimiterIDPCorrection end
+
+function (limiter!::SubcellLimiterIDPCorrection)(u_ode,
+                                                 integrator::Trixi.SimpleIntegratorSSP,
+                                                 stage)
+    limiter!(u_ode, integrator.p, integrator.t, integrator.dt,
+             integrator.p.solver.volume_integral)
+end
```

Comment:
```suggestion
    semi = integrator.p
    limiter!(u_ode, semi, integrator.t, integrator.dt,
             semi.solver.volume_integral)
end
```
To clarify that you unpack the semidiscretization